i18n/models.py

```diff
@@ -62,10 +69,13 @@ def gather_strings(cls):
             if obj.should_be_translated:
                 key = obj.i18n_key
                 strings[key] = {}
+                serialized_obj = {}
+                if Serializer:
+                    serialized_obj = Serializer(obj).data
                 for field in cls.internationalizable_fields():
-                    string = getattr(obj, field)
+                    string = serialized_obj[field] if serialized_obj.has_key(field) else getattr(obj, field)
```

Comment:
I wonder if this gets cleaner by creating serializers for _all_ models, and returning the serialized dictionary filtered by `internationalizable_fields`?

Reply:
yeah, I could definitely see that!

Rebrand "gather_strings" to "serialize i18n data" (which is a more accurate and descriptive name, anyway) and do something like require you to define serializers rather than `internationalizable_fields`, or perhaps something that can easily generate serializers for the simple use cases but would let you manually define them for things like this that are doing something a little more complicated.

If I had remembered that django serializers were a thing, I probably would have built it that way to begin with.